tests/pytests/test_gauxc_writer/gauxc_writer.py

```diff
@@ -0,0 +1,212 @@
+import h5py
+import numpy as np
+import os
+import psi4
+
+def write_results(run_name, wfn):
+    try:
+        #== create reference ==#
+        with h5py.File(os.path.join(os.path.dirname(__file__), f'{run_name}.hdf5'), "w") as f:
+            #== write molecule info ==#
+            wfn_molecule = wfn.molecule()
+
+            molecule_dtype = {
+                'names': ['Atomic Number', 'X Coordinate', 'Y Coordinate', 'Z Coordinate'], 
+                'formats': ['<i8', '<f8', '<f8', '<f8'], 
+                'offsets': [0, 8, 16, 24], 
+                'itemsize': 32
+            }
+           
+            molecule_dset = f.create_dataset("/MOLECULE", shape=(wfn_molecule.natom(),), dtype=molecule_dtype)  
+          
+            for iatom in np.arange(0, wfn_molecule.natom()):
+                atom_data = ( int(wfn_molecule.Z(iatom)), wfn_molecule.x(iatom), wfn_molecule.y(iatom), wfn_molecule.z(iatom) )
+                molecule_dset[iatom] = atom_data 
+          
+            #== write basis info ==#
+            basis = wfn.basisset()
+            basis_dtype = {
+                'names': ['NPRIM', 'L', 'PURE', 'ALPHA', 'COEFF', 'ORIGIN'], 
+                'formats': ['<i4', '<i4', '<i4', ('<f8', (16,)), ('<f8', (16,)), ('<f8', (3,))], 
+                'offsets': [0, 4, 8, 16, 272, 528], 
+                'itemsize': 552
+            }
+            
+            basis_dset = f.create_dataset("/BASIS", shape=(basis.nshell(),), dtype=basis_dtype)  
+          
+            for ishell in np.arange(0, basis.nshell()):
+                basis_data = [0, 0, 0, 
+                  np.zeros(16, dtype=np.float64), 
+                  np.zeros(16, dtype=np.float64), 
+                  np.zeros(3, dtype=np.float64)
+                ]
+                
+                psi4_shell = basis.shell(ishell)
+          
+                basis_data[0] = psi4_shell.nprimitive
+                basis_data[1] = psi4_shell.am
+                basis_data[2] = 0 if any([ psi4_shell.is_cartesian() ]) else 1
+           
+                for iprim in np.arange(0, psi4_shell.nprimitive): 
+                  basis_data[3][iprim] = psi4_shell.exp(iprim)
+                  basis_data[4][iprim] = psi4_shell.coef(iprim)
+          
+                for icoord in np.arange(0,3):
+                  basis_data[5][icoord] = psi4_shell.coord(icoord)
+          
+                basis_dset[ishell] = tuple(basis_data)
+        
+            initial_index = np.arange(basis.nbf())
+            final_index = np.zeros(basis.nbf(), dtype=int)
+            
+            max_am = 7;
```

Comment:
Caught this just now; I suppose this is another area where I use a hard-coded AM value to handle construction of the ERI permutation matrix. Admittedly not immediately sure of the best way to handle this case, since this is in a script not directly connected to the build system; but I can ponder that.

Reply:
There should be a Python interface to GauXC that could be used to query the maximum angular momentum.

Reply:
@susilehtola If we were to do a python interface for GauXC (which is on the roadmap), we'd want to do the full thing at once, which will take some significant effort. I wouldn't hold this up for that. We can add a runtime checker for this value (see other comment) and Psi4 can add a python function that wraps it if it needs to be queried.

Reply:
@wavefunction91 yeah that's what I was thinking too

Reply:
@wavefunction91 Yeah, this is an acceptable long-term solution to me, as well. As for a more short-term solution, I'm looking at the possibility of adding some sort of `gauxc_writer.py.in` file that, when Psi4 is built, gets configured into a `gauxc_writer.py` file with the correct maximum AM value.

Reply:
But that again introduces a silent dependency on the exact flavor that GauXC was compiled with. If GauXC gets updated in the background with a different version, Psi4 will crash.

Reply:
@susilehtola That's true, but I'm thinking of it as a short-term solution since there is no GauXC Python interface currently. Is it better if we just keep it as hard coded for now?

Reply:
fwiw, L2 has the same weakness of not being hot-swappable against libraries with different AM configuration.

Reply:
@susilehtola So I suppose the best thing to do for now is to wait for the runtime GauXC AM checker to get added, and move forward with @wavefunction91's suggestion, then?

Reply:
https://github.com/wavefunction91/GauXC/pull/126

Reply:
Excellent!

Reply:
> fwiw, L2 has the same weakness of not being hot-swappable against libraries with different AM configuration.

@loriab believe me, I know....

Reply:
Okay all (@wavefunction91 @susilehtola @loriab)! I have added runtime determination of GauXC AM support everywhere where it is needed, both in the main code, and in the test file. I've also updated the code to pin to the GauXC commit for which this functionality is added. I'll resolve all other comment chains regarding this issue except this one; of course, feel free to let me know if you'd like to see changes in how I handle runtime AM checking on the Psi4 side.